test/unit/fapi-json.c

```diff
@@ -101,6 +101,79 @@ static void ifapi_cleanup_policy(TPMS_POLICY *policy)
     }
 }
 
+/* 6 copies of cleanup functions from ifapi_keystore.c */
+
+void
+cleanup_ifapi_duplicate(IFAPI_DUPLICATE * duplicate) {
+    if (duplicate != NULL) {
+        SAFE_FREE(duplicate->certificate);
+    }
+}
+
+void
+cleanup_ifapi_hierarchy(IFAPI_HIERARCHY * hierarchy) {
+    if (hierarchy != NULL) {
+        SAFE_FREE(hierarchy->description);
+    }
+}
+
+void
+cleanup_ifapi_ext_pub_key(IFAPI_EXT_PUB_KEY * key) {
+    if (key != NULL) {
+        SAFE_FREE(key->pem_ext_public);
+        SAFE_FREE(key->certificate);
+    }
+}
+
+void
+cleanup_ifapi_key(IFAPI_KEY * key) {
+    if (key != NULL) {
+        SAFE_FREE(key->policyInstance);
+        SAFE_FREE(key->serialization.buffer);
+        SAFE_FREE(key->private.buffer);
+        SAFE_FREE(key->description);
+        SAFE_FREE(key->certificate);
+        SAFE_FREE(key->appData.buffer);
+    }
+}
+
+void
+cleanup_ifapi_nv(IFAPI_NV * nv) {
+    if (nv != NULL) {
+        SAFE_FREE(nv->serialization.buffer);
+        SAFE_FREE(nv->appData.buffer);
+        SAFE_FREE(nv->policyInstance);
+        SAFE_FREE(nv->description);
+        SAFE_FREE(nv->event_log);
+    }
+}
+
+void
+cleanup_ifapi_object(
+    IFAPI_OBJECT * object)
+{
+    if (object != NULL) {
+        if (object->objectType != IFAPI_OBJ_NONE) {
```

Comment:
Just a nit-pick - if you would combine the two if statements into one:
if (object != NULL && object->objectType != IFAPI_OBJ_NONE) {
...
it would save you one level of indentation.